### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jBosak98**
Admin panel init
Backend:
rest api in accordance with mobile app
authorization
authentication
rest api to admin panel

### CODE REVIEW

**Comment by Burdzi0** [comment]
+19,484 −26
Are you kidding me????
Who is going to look at all these changes? 
I don't have a free month

**Comment by jBosak98** [comment]
well, over 17 000 lines come from package-json which you don't need to look at.

File: backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    private final EventService eventService;
+    private final PlaceService placeService;
+    private final PathService pathService;
+    private final TimetableService timetableService;
+    private final SpeakerService speakerService;
+    private final LastUpdateService lastUpdateService;
+
+    private final ValidationErrorService validationErrorService;
+
+    public AppDataController(
```
Comment (Burdzi0) [inline backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java]: This is wrong, injecting more than 3 services is a sign of wrong architecture. Is it necessary?

File: backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                .getContext()
+                .getAuthentication()
+                .getAuthorities()
+                .stream()
+                .map(GrantedAuthority::getAuthority)
+                .collect(Collectors.toSet());
+        if (roles.contains("ROLE_USER")){
+            appData = new AppData(
+                    eventService.findActiveEvents(),
+                    placeService.findAllPlaces(),
```
Comment (Burdzi0) [inline backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java]: `findAllPlaces()` is ok? Every single other one contains "Active" keyword

Comment (jBosak98) [reply backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java]: Yes, the majority of models are dependent of "active" located in path. I guess that only places are independent.

File: backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java (hunk 1 above)
Comment (jBosak98) [reply backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java]: I agree, but I don't know how to handle it due to we need to return Appdata which is composed of five models.

Comment (Burdzi0) [reply backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java]: The problem is apparently in `AppData.java` design. Passing 5 list objects in constructor is a bad move and you know it. That's why you have to have 7 fields in constructor in `AppDataController.java`. Do you need all the data available at once? Does the mobile app requires it?

Comment (jBosak98) [reply backend/src/main/java/pl/wroclaw/asi/labdaybackendspring/controllers/AppDataController.java]: refactored here: https://github.com/asi-pwr/LabDayBackendSpring/pull/24

**Comment by jBosak98** [comment]
2 days without issues or comments, so I merge it

**Pull request merged**